src/pipelinedefinition.cpp

```diff
@@ -546,8 +546,8 @@ class NodeValidator {
         // Affect shape by demultiplexer/gather if applies.
         const auto& tensorInput = this->inputsInfo.at(modelInputName);
         const auto& tensorOutput = this->dependencyOutputsInfo.at(modelOutputName);
-        shape_t tensorInputShape = tensorInput->getShape();
-        shape_t tensorOutputShape = tensorOutput->getShape();
+        shape_t tensorInputShape = tensorInput->getShape_3().getFlatShape();
+        shape_t tensorOutputShape = tensorOutput->getShape_3().getFlatShape();
```

Comment:
I'd also prefer not to use `getFlatShape` here. We could check if the shape is dynamic and abort immediately. `getFlatShape` will be removed in cleanup state and we should not rely on it or move this task to future self!